### PR TITLE
Add redis standard function and improve the handling of error

### PR DESCRIPTION
**Pull request by akiidjk**
### Update 0
- Add model to describe data
- Add a simple function to communicate with redis
- Add an error manager to handle errors in the code

-----------

### Update 1

- Generalised the get and set function (redis) with two macros
- Add a space in a comment in the model file (this because the compile phase in the github action generates an error lol)

### CODE REVIEW

**Review by danmig06 — CHANGES_REQUESTED** [review]
Looks alright, you also got rid of other boiler plate stuff we had missing, the redis GET/SET requests are easly generalizable though, just look at how many lines repeat in each of the 4 functions.

File: backend/src/redis.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// File with redis call
```
Comment (danmig06) [inline backend/src/redis.rs:1]: Make this code more D.R.Y. (read the comments)

File: backend/src/error_manager.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+use thiserror::Error;
```
Comment (danmig06) [inline backend/src/error_manager.rs:1]: This is nice

File: backend/src/main.rs [hunk 3]
```diff
@@ -1,47 +1,58 @@
 mod data;
```
Comment (danmig06) [inline backend/src/main.rs:1]: This will have to be changed in regards to the changes requested (read the comments)

**Comment by danmig06** [comment]
What you could do is adding enum types for the type of request and, if you really want a separate _json function, making a macro that does the conversion and passes it the function along with the command SET

**Comment by akiidjk** [comment]
OK so I try to generalise even more and repeat as little code as possible, I'll work on it this evening

**Review by danmig06 — APPROVED** [review]
this is more readable, consider using enum values for commands and making just one function called redis_send_command for example

**Pull request merged**